src/main/scala/cromwell/engine/db/slick/DataAccessController.scala

```diff
@@ -0,0 +1,310 @@
+
+        // Check each symbol to see if one exists with the (scope, name, iteration) combo
+        val symbolSelect = workflowExecutionSelect flatMap { workflowExecutionRow =>
+          dataAccess.symbols.filter(symbolRow =>
+            symbolRow.workflowExecutionId === workflowExecutionRow.workflowExecutionId &&
+              symbolRow.scope === symbol.key.scope &&
+              symbolRow.name === symbol.key.name &&
+              symbolRow.iteration === symbol.key.iteration
+          ).result.headOption
+        }
+
+        symbolSelect map {
+          case Some(symbolRow) =>
+            // Select found a row. Update.
+            val query = dataAccess.symbols.filter(_.symbolId === symbolRow.symbolId)
+            val columns = query.map(_.wdlValue) // NOTE: Not allowing updating of the type, just the value
+            columns.update(symbol.wdlValue.map(wdlValueToString).orNull)
+          case None =>
+            // Select did not find a row. Insert.
+            workflowExecutionSelect flatMap { workflowExecutionRow =>
+              dataAccess.symbolsAutoInc += new Symbol(workflowExecutionRow.workflowExecutionId.get,
+                symbol.key.scope, symbol.key.name, symbol.key.iteration,
+                if (symbol.isInput) "INPUT" else "OUTPUT",
+                symbol.wdlType.toString, symbol.wdlValue.map(wdlValueToString).orNull)
+            }
+        }
+
+      }
+    }
+
+    val allInsertsOrUpdates = DBIO.sequence(callInsertsOrUpdates.toSeq ++ symbolInsertOrUpdates.toSeq.flatten)
+    val allInsertsOrUpdatesFuture = database.run(allInsertsOrUpdates.transactionally)
+    Await.result(allInsertsOrUpdatesFuture, Duration.Inf)
+  }
+
+  override def updateWorkflow(id: WorkflowId, state: WorkflowState): Unit = {
+    val updateFuture: Future[Int] = database.run((for {
+      workflowExecutionSelect <- dataAccess.workflowExecutions.filter(_.workflowExecutionUuid === id.toString)
+    } yield workflowExecutionSelect.status).update(state.toString).transactionally)
+    // TODO: "Assert" only one row updated?
+    Await.result(updateFuture, Duration.Inf)
+  }
+
+  override def query(workflowId: Option[Seq[WorkflowId]], wdlUris: Option[Seq[String]],
+                     states: Option[Seq[WorkflowState]], beforeStart: Option[Date],
+                     afterStart: Option[Date], beforeEnd: Option[Date], afterEnd: Option[Date]):
+  Seq[QueryWorkflowExecutionResult] = {
+
+    // TODO: Another way to construct a query without an empty filter?
+    var query = dataAccess.workflowExecutions.filter(row => true: Rep[Boolean])
+
+    if (workflowId.isDefined) {
+      query = query.filter { row =>
+        val reps = workflowId.get.map { id =>
+          row.workflowExecutionUuid === id.toString
+        }
+        // Modified from: http://slick.typesafe.com/doc/3.0.0/queries.html#sorting-and-filtering
+        reps.reduceLeftOption(_ || _).getOrElse(true: Rep[Boolean])
+      }
+    }
+
+    if (wdlUris.isDefined) {
```

Comment:
Look at Mr Fancy Pants!

Agree 100% that that would be slick (I'm a punny guy) and bonus points would be awarded if one took things to that level.